autotst/reaction.py

```diff
@@ -311,49 +312,64 @@ def get_labeled_reaction(self):
         assert (
             self.label or self.rmg_reaction), "You must provide a reaction or a reaction label"
 
-        match = False
-        if self.label:
-            rmg_reactants = []
-            rmg_products = []
-            r, p = self.label.split("_")
-            for react in r.split("+"):
-                s = RMGMolecule(smiles=react)
-                rmg_reactants.append(s)
-            for prod in p.split("+"):
-                s = RMGMolecule(smiles=prod)
-                rmg_products.append(s)
-
-            test_reaction = RMGReaction(
-                reactants=rmg_reactants, 
-                products=rmg_products)
-
-            if self.rmg_reaction:
-                assert self.rmg_reaction.is_isomorphic(
-                    test_reaction), "The reaction label provided does not match the RMGReaction provided..."
-
+        match = False # We have not found a labeled reaction that matches the one provided.
+        if self.label: # A reaction label was provided
+            
+            # Generating lists of lists. Main list is the reactans or products
+            # Secondary list is composed of the resonance structures for that species
+            r, p = self.label.split("_") 
+            rmg_reactants = [RMGMolecule(smiles=smile).generate_resonance_structures() for smile in r.split("+")]
+            rmg_products = [RMGMolecule(smiles=smile).generate_resonance_structures() for smile in p.split("+")]
+
+            combos_to_try = list(itertools.product(
+                itertools.product(*rmg_reactants),
```

Comment:
cast these itertools.product calls to a `list` here?
eg. `list(itertools.product(*rmg_reactants)),` and `list(itertools.product(*rmg_products))`

Reply:
Okay, this makes sense to add `list` when generating the `combos_to_try`.